tests/adapter/dbt/tests/adapter/query_comment/fixtures.py

```diff
@@ -0,0 +1,64 @@
+MACROS__MACRO_SQL = """
+{%- macro query_header_no_args() -%}
+{%- set x = "are pretty cool" -%}
+{{ "dbt macros" }}
+{{ x }}
+{%- endmacro -%}
+
+
+{%- macro query_header_args(message) -%}
+  {%- set comment_dict = dict(
+    app='dbt++',
+    macro_version='0.1.0',
+    dbt_version=dbt_version,
+    message='blah: '~ message) -%}
+  {{ return(comment_dict) }}
+{%- endmacro -%}
+
+
+{%- macro ordered_to_json(dct) -%}
+{{ tojson(dct, sort_keys=True) }}
+{%- endmacro %}
+
+
+{% macro invalid_query_header() -%}
+{{ "Here is an invalid character for you: */" }}
+{% endmacro %}
+
+"""
+
+MODELS__X_SQL = """
+{% set blacklist = ['pass', 'password', 'keyfile', 'keyfile.json', 'password', 'private_key_passphrase'] %}
+{% for key in blacklist %}
+  {% if key in blacklist and blacklist[key] %}
+      {% do exceptions.raise_compiler_error('invalid target, found banned key "' ~ key ~ '"') %}
+  {% endif %}
+{% endfor %}
+
+{% if 'type' not in target %}
+  {% do exceptions.raise_compiler_error('invalid target, missing "type"') %}
+{% endif %}
+
+{% set required = ['name', 'schema', 'type', 'threads'] %}
+
+{# Require what we docuement at https://docs.getdbt.com/docs/target #}
+{% if target.type == 'postgres' or target.type == 'redshift' %}
```

Comment:
The problem with doing it this way is that it doesn't support external adapter maintainers very well, and we don't actually provide an example of how to get this working with an external adapter except by replacing this entire model. In addition, putting the test in jinja model code is kind of a bad pattern.

 It would be better to put the required fields in some kind of config that can be retrieved. Also, it would probably be better to move the checks into the actual test code, rather than embedding it in the model. I'd make the "required" fields a fixture that can be overridden in the adapter specific tests,  and then pull that into the test method and test for the existence of those fields in the target in the test.  I'm not exactly sure how you can get the target dictionary. Possibly from the adapter? I can help look for how to get that if you need.

Reply:
I'm not even sure why we're doing this in this test. What does this have to do with query comments?

Reply:
as we talked about i'm a little hazy on why this is here other than validation logic possibly against if we did import it to a different database to make sure its not actually relying off say postgres. 

would love to find a better way to work on this will play around some of these other options if we do think this chucnk is needed at all.

Reply:
Hey @gerda had a few questions/ideas are we able to set up some global stuff for checks in a fixture that can't be overwritten by chance? like the 
`{% set required = ['name', 'schema', 'type', 'threads'] %}` that might not change between the adapters?

trying to get ahold of the related information that would be same as `target` in the fixture looks like from our `TestProjInfo` something like `profiles_dir` would contain the information we would want just not sure if theres a good way to tap into that possibly by some `os` read stuff?